backends/lean/Base/Primitives/Scalar.lean

```diff
@@ -1419,6 +1419,19 @@ instance (ty: ScalarTy) : LinearOrder (Scalar ty) where
     right; exact (Scalar.le_equiv _ _).2 H
   decidableLE := ScalarDecidableLE ty
 
+-- Coercion theorems
+-- This is helpful whenever you want to "push" casts to the innermost nodes
+-- and make the cast normalization happen more magically.
+
+@[simp, norm_cast]
+theorem coe_max {ty: ScalarTy} (a b: Scalar ty): Max.max a b = (Max.max a b: ℤ) := by
+  -- TODO: redo me better because this is a bit weird.
```

Comment:
Redo what? The proof?

Reply:
Yeah, I was not very happy with the body of the proof, I need to take a look again.

Reply:
Maybe we can merge this for the time being and revisit that proof later or I can remove the TODO, let me know what you prefer